scene/resources/skeleton_modification_2d.cpp

```diff
@@ -0,0 +1,3433 @@
+
+	ClassDB::bind_method(D_METHOD("get_is_setup"), &SkeletonModificationStack2D::get_is_setup);
+
+	ClassDB::bind_method(D_METHOD("set_enabled", "enabled"), &SkeletonModificationStack2D::set_enabled);
+	ClassDB::bind_method(D_METHOD("get_enabled"), &SkeletonModificationStack2D::get_enabled);
+
+	ClassDB::bind_method(D_METHOD("set_strength", "strength"), &SkeletonModificationStack2D::set_strength);
+	ClassDB::bind_method(D_METHOD("get_strength"), &SkeletonModificationStack2D::get_strength);
+
+	ClassDB::bind_method(D_METHOD("get_skeleton"), &SkeletonModificationStack2D::get_skeleton);
+
+	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "enabled"), "set_enabled", "get_enabled");
+	ADD_PROPERTY(PropertyInfo(Variant::FLOAT, "strength", PROPERTY_HINT_RANGE, "0, 1, 0.001"), "set_strength", "get_strength");
+	ADD_PROPERTY(PropertyInfo(Variant::INT, "modification_count", PROPERTY_HINT_RANGE, "0, 100, 1"), "set_modification_count", "get_modification_count");
+}
+
+SkeletonModificationStack2D::SkeletonModificationStack2D() {
+	skeleton = nullptr;
+	modifications = Vector<Ref<SkeletonModification2D>>();
+	is_setup = false;
+	enabled = false;
+	modifications_count = 0;
+	strength = 1;
+}
+
+///////////////////////////////////////
+// Modification2D
+///////////////////////////////////////
+
+void SkeletonModification2D::execute(float delta) {
+	if (get_script_instance()) {
+		if (get_script_instance()->has_method("execute")) {
+			get_script_instance()->call("execute", delta);
+		}
+	}
+
+	if (!enabled)
+		return;
+}
+
+void SkeletonModification2D::setup_modification(SkeletonModificationStack2D *p_stack) {
+	stack = p_stack;
+	if (stack) {
+		is_setup = true;
+	} else {
+		WARN_PRINT("Could not setup modification with name " + get_name());
+	}
+
+	if (get_script_instance()) {
+		if (get_script_instance()->has_method("execute")) {
+			get_script_instance()->call("setup_modification", p_stack);
+		}
+	}
+}
+
+void SkeletonModification2D::draw_editor_gizmo() {
+	if (get_script_instance()) {
+		if (get_script_instance()->has_method("draw_editor_gizmo")) {
+			get_script_instance()->call("draw_editor_gizmo");
+		}
+	}
+}
+
+bool SkeletonModification2D::_print_execution_error(bool p_condition, String p_message) {
+	if (!is_setup) {
+		return p_condition;
+	}
+
+	if (p_condition && !execution_error_found) {
+		ERR_PRINT(p_message);
+		execution_error_found = true;
+	}
+	return p_condition;
+}
+
+void SkeletonModification2D::set_enabled(bool p_enabled) {
+	enabled = p_enabled;
+
+#ifdef TOOLS_ENABLED
+	if (editor_draw_gizmo) {
+		if (stack) {
+			stack->set_editor_gizmos_dirty(true);
+		}
+	}
+#endif // TOOLS_ENABLED
+}
+
+bool SkeletonModification2D::get_enabled() {
+	return enabled;
+}
+
+float SkeletonModification2D::clamp_angle(float angle, float min_bound, float max_bound, bool invert) {
+	// Map to the 0 to 360 range (in radians though) instead of the -180 to 180 range.
+	if (angle < 0) {
+		angle = (Math_PI * 2) + angle;
```

Comment:
You could use `angle += Math_TAU;`.

Why do this only if `angle` is negative? If I don't go wrong, you should always shift the entire angle by `Math_PI`.

Reply:
Cool, I'll check and if `Math_TAU` gives the same results I'll use that.

As to when it is only done when negative, this is because we need to convert negative angles to the positive range, so it goes from `0 to 360` instead of  `-180 to 180`, which is how Godot angles are currently. Angles over `180` are flipped into the negative range, starting at `-180` and ending at `0`, and that is what this code is for and why it only checks for negatives.
If we added Math_PI to the entire thing, then the `0 to 180` range would be `180 to 360`, which is incorrectly mapped and would mess up the clamping.